src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        str += separator;
+        continue;
+      }
+
+      if (i !== this.length - 1) {
+        str += this[i] + '' + separator;
+      } else {
+        str += this[i];
+      }
```

Comment:
```suggestion
      if (this[i] !== null || this[i] !== undefined) {
        str += this[i];
      }

      if (i < this.length - 1) {
        str += separator;
      }
```

Look, here you can make your code work faster, using 2 checks instead of 3 (and not using `continue`).
First - you check if your element is not `null` or `undefined`. If so - you add this element to the string. Then you check if it's not the last element (i < this.length - 1), and if so - add separator.
If your element is `null` or `undefined` - you just add separator (if it's not the last element in array)